docs/source/ucx.rst

```diff
@@ -1,35 +1,87 @@
+
+- ``ucx.tcp: true`` -- **required.**
+
+    Adds ``tcp`` to ``UCX_TLS``, enabling TCP transfers over UCX if NVLink or InfiniBand are unavailable or disabled.
+    *Why is this required if they are available and enabled?*
+
+- ``ucx.nvlink: true`` -- required for NVLink.
+
+    Adds ``cuda_ipc`` to ``UCX_TLS``, enabling NVLink transfers over UCX; affects intra-node communication only.
 
+- ``ucx.infiniband: true`` -- required for InfiniBand.
 
-- ``DASK_UCX__CUDA_COPY=True`` (default: ``False``): *Always required for UCX*, adds ``cuda_copy`` to ``UCX_TLS`` -- required for all CUDA transfers in UCX, both intra- and inter-node;
-- ``DASK_UCX__TCP=True`` (default: ``False``): *Always required for UCX*, adds ``tcp`` to ``UCX_TLS`` -- required for all TCP transfers (e.g., where NVLink or IB is not available/disabled) in UCX, both intra- and inter-node;
-- ``DASK_UCX__NVLINK=True`` (default: ``False``): Adds ``cuda_ipc`` to ``UCX_TLS`` -- required for all NVLink transfers in UCX, only affects intra-node.
-- ``DASK_UCX__INFINIBAND=True`` (defalt: ``False``): Adds ``rc`` to ``UCX_TLS`` -- required for all InfiniBand transfers in UCX, only affects inter-node.
-- ``DASK_UCX__RDMACM=True`` (default: ``False``): Replaces ``sockcm`` by ``rdmacm`` in ``UCX_TLS`` and ``UCX_SOCKADDR_TLS_PRIORITY``. ``rdmacm`` is the recommended method by UCX to use with IB and currently won't work if ``DASK_UCX__INFINIBAND=False``.
-- ``DASK_UCX__NET_DEVICES=mlx5_0:1`` (default: ``None``, causes UCX to decide what device to use, possibly being suboptimal, implies ``UCX_NET_DEVICES=all``): this is very important when ``DASK_UCX__INFINIBAND=True`` to ensure the scheduler is connected over the InfiniBand interface. When ``DASK_UCX__INFINIBAND=False`` it's recommended to use the ethernet device instead, e.g., ``DASK_UCX__NET_DEVICES=enp1s0f0`` on a DGX-1.
-- ``DASK_RMM__POOL_SIZE=1GB``: allocates an RMM pool for the process. In some circumstances, the Dask scheduler will deserialize CUDA data and cause a crash if there's no pool.
+    Adds ``rc`` to ``UCX_TLS``, enabling InfiniBand transfers over UCX; affects inter-node communication only.
+
+
+- ``ucx.rdmacm: true`` -- recommended for InfiniBand.
+
+    Replaces ``sockcm`` with ``rdmacm`` in ``UCX_TLS`` and ``UCX_SOCKADDR_TLS_PRIORITY``, *enabling remote direct memory access (RDMA) for connection management.*
+    This is recommended by UCX for use with InfiniBand, and will have no effect if InfiniBand tranfers are disabled.
+
+- ``ucx.net-devices: <str>`` -- recommended.
+
+    Explicitly sets ``UCX_NET_DEVICES`` instead of defaulting to ``"all"``, which can result in suboptimal performance.
+    If using InfiniBand, set to the desired IB device, e.g. ``"mlx5_0:1"``.
+    If InfiniBand is disabled, set to the ethernet device, e.g. ``"enp1s0f0"`` on a DGX-1.
+    All available UCX-compatible devices can be listed by running ``ucx_info -d`` or ``ifconfig`` *(are all the options under ifconfig compatible)*.
```

Comment:
I don't think all options under `ifconfig` are valid, for example `veth*` (often used by Docker) aren't. `ucx_info -d` only should be more reliable.

Also, interestingly the most important use case for us is missing in this section (it was missing before your proposed changes here too), which is the `"auto"` option, as documented in https://github.com/rapidsai/dask-cuda/blob/e72d776645ee3ee7e1672f73c9c043731cf401d5/dask_cuda/cli/dask_cuda_worker.py#L187-L203 and https://github.com/rapidsai/dask-cuda/blob/e72d776645ee3ee7e1672f73c9c043731cf401d5/dask_cuda/local_cuda_cluster.py#L72-L86 . Could you add that here as well? It's documented in "Important Notes" in this file below, but it seems sensible to have it here too.

Reply:
It does make sense to include information about `"auto"` here. Are there any resources on installing UCX-Py with hwloc support? Didn't see anything explicit about it in the installation guide.

Reply:
It's in the build dependencies: https://ucx-py.readthedocs.io/en/latest/install.html#build-dependencies

Reply:
Question about specifying an interface when InfiniBand is disabled:

I notice later on in the documentation, we say when setting up a scheduler for UCX that setting `DASK_UCX__NET_DEVICES` can generally be skipped when using a non-InfiniBand interface is used. Is this universally true, or does it only apply to the scheduler?

Reply:
This can apply to all processes. `UCX_NET_DEVICES` is merely a way to limit the visibility of network devices to UCX, meaning it won't "see" any others. We can still choose the device by specifying `--interface` to scheduler and workers, which is generally preferred if no IB is used. To be even more explicit, we're likely going to drop `DASK_UCX__NET_DEVICES` entirely starting from UCX 1.10, because UCX is now capable of ensuring it uses the IB interface topologically closest to the GPU (which is what `"auto"` does now).

Reply:
In this case, should I remove the recommendation to explicitly set `ucx.net-devices` to an ethernet device when InfiniBand is disabled? Seems like an inconsistency, unless I'm missing something.

Reply:
The way I read is we recommend users to set to the "optimal" device, so I would still leave it in place. I know in the past we had issues with suboptimal devices being chosen and it seems like UCX 1.10+ is better in that regard, so what I meant to say was that starting from UCX 1.10 we'll then prefer not to set `UCX_NET_DEVICES`, but I don't think in our current state (UCX 1.9), we're ready for that.